src/terraform_module_generator/schema/blocks/block.py

```diff
@@ -0,0 +1,219 @@
+"""
+    :copyright: 2022 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+import textwrap
+import typing
+
+from inmanta_module_factory import builder, inmanta
+from inmanta_module_factory.helpers.utils import inmanta_entity_name, inmanta_safe_name
+
+from terraform_module_generator.schema import const, mocks
+from terraform_module_generator.schema.attributes.base import Attribute
+from terraform_module_generator.schema.helpers.cache import cache_method_result
+from terraform_module_generator.schema.mocks.nested_block import NestedBlockMock
+
+if typing.TYPE_CHECKING:
+    from terraform_module_generator.schema.blocks.nested_block import NestedBlock
+
+
+class Block:
+    def __init__(self, name: str, path: typing.List[str], schema: typing.Any) -> None:
+        self.name = name
+        self.path = path
+
+        nested_blocks: typing.List[mocks.NestedBlockMock] = []
+        self.attributes: typing.List[Attribute] = []
+
+        for attribute in Block.get_attributes(path + [inmanta_safe_name(name)], schema):
+            # For each of the attributes of this block, instead of simply considering
+            # them as attributes, we first check if some of them could be converted to
+            # entities.  This is only the case if the attribute type is a structure, or
+            # a collection of structures.  In those case, instead of adding the attribute
+            # as an attribute to the block, we will build a nested block and add it to
+            # this block.
+            try:
+                nested_blocks.append(attribute.as_nested_block())
+            except NotImplementedError:
```

Comment:
Not a style I'm used to, but very pythonic. 

cool.

Reply:
Me neither, but I kind of like it, I will experiment more with that